.github/workflows/compliance.yml

```diff
@@ -0,0 +1,140 @@
+#####################################################################################################################
+# @author       Michael Smith
+# @date         2021-08-23
+# @description  As SFDO Managed Packages start using the newly introduced Instrumentation Services to 'log' 
+#               usage data to Splunk and Argus, this Compliance Check action does two things:
+#               1. Ensures that Packaged Apex Classes do not make any direct reference to the core SfdoLogUtils class.
+#                  Any code references to this class will cause the Compliance build to fail.
+#               2. At least in the short term, adds an Instrumentation reviewer to any PR that in any way references or
+#                  or uses the SfdoInstrumentationService. This is primarily meant as a short term action to aid in
+#                  adoption of the new instrumentation services.
+#####################################################################################################################
+name: Instrumentation_Service_Compliance_Verification
+on:
+  pull_request:
+    # Filter the job to only execute if CLS files are in the PR
+    paths:    
+      - '**.cls'
+      - '!unpackaged/**.cls'
+
+jobs:
+  Code_Check:
+
+    # Constant vars
+    # These should be adjusted per repository
+    env:
+      DEFAULT_REVIEWER: force2b
+      ALTERNATE_REVIEWER: lparrott
+      ASSIGNED_LABEL: Instrumentation
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
We need to find a more efficient way to get the references we need. Fetching the entire history for HEAD is pretty inefficient: it looks like your workflow runs are averaging around 20-25s, but 18s of that is spent on this step. 18s will add up very quickly in NPSP.

I'm okay with leaving this for now, but we'll want to revisit this before merging into 234. Once you have a known good workflow you can try tweaking this value to save time. We may be able to sidestep it by retrieving and parsing the [PR diff url](https://github.com/SalesforceFoundation/NPSP/pull/6623.diff) (`github.event.pull_request.diff_url`), for example.

Reply:
Good point. I did notice that was taking some time. Maybe once this is working fully, a incremental check would be sufficient - just checking the latest commit contents rather than the full diff?

Reply:
I see both of those return the same result, but I'm not sure a) what the difference is; or if one is faster than the other. Also, should I change or remove `fetch-depth`? I don't think there's a way to test that at the command line though.